### ISSUE TITLE
Provide flag to make access request `reason` mandatory

### ISSUE THREAD

**Issue by kimlisa** [gravitational/teleport#7560]
### What
Currently, there is no config available to allow access request `reasons` to be mandatory, other than to enable `waiting room` as shown below:
```
options:
  request_access: reason
  request_prompt: <some custom prompt to show in reason dialogue>
```
With this option, the request reasons are mandatory, but requires user to go through the waiting room upon first login.

### Why
Customer request

**Comment by ArunNadda** [comment]
Customer want `request reason` mandatory when requesting for extra roles, but they dont want users to wait during login to teleport (and get basic access).

**Comment by ArunNadda** [comment]
Regarding question `https://goteleport.com/docs/enterprise/workflow/#example-setup. Does options.request_acces: reason not work?`

As mentioned in GH ticket too, option request_access: reason causes users to request the roles automatically at even first login to Teleport cluster and land them in waiting room (as in snapshot from UI)... or in CLI as below
here i am trying to login as contractor user which is member of contractor role and has request-reason set to reason. Even though i did not request for access role specifically, its still requested at first login
% tsh login --proxy=hatelcluster.aknteleport.cf:443 --user=contractor --request-reason=test
Enter password for Teleport user contractor:
Enter your OTP token:
731173

Creating request...
Request ID: ca37ce74-83a4-4f12-9940-f4155a827e57
Username:   contractor
Roles:      access
Reason:     "test"
Reviewers:  [none] (suggested)
Status:     PENDING

hint: use 'tsh login --request-id=<request-id>' to login with an approved request

Waiting for request approval...
Ask is to give users login access to teleport without any waiting room, so they can use their own roles/permissions. And if they are requesting any new roles, reason should be mandatory at that point..

![request_wating_at_first_login](https://user-images.githubusercontent.com/9428444/162361356-6c229b9f-0568-4952-a890-0308ac50cf48.png)

**Comment by klizhentas** [comment]
Assuming that we add additional knob to provide a reason. What prevents users from supplying insufficient reason? In this case the request should be rejected, just as it should be rejected with no provided reason.

In this case, it's unclear why optional reason would differ from invalid reason for compliance purposes.

**Comment by hyphen-jaydenzheng** [comment]
@klizhentas We used the integration with Jira, but the solution lacks the error message or the reason to be shown in the Teleport UI. When Jira rejects the request and sends the request back to the Teleport-Plugin, it simply just shows the access request was denied. The end-user might get confused about this. 

Currently, we have implemented a slackbot approach to display the reject reason to the user. If we can let Teleport handle this from the start (enforce reason) then we can save the trouble of maintaining the pipeline of notice to the end user via the slack bot; equal less work for clients.
